src/bundle/Resources/public/js/scripts/admin.location.visibility.js

```diff
@@ -1,12 +1,36 @@
-(function (global, doc) {
-    const updateVisibility = event => {
-        event.target.closest('.ez-checkbox-icon').classList.toggle('is-checked');
+(function(global, doc) {
+    const SELECTOR_VISIBILITY_CHECKBOXES = '#ez-tab-location-view-locations .ez-checkbox-icon__checkbox';
+    const fetchInit = {
+        credentials: 'same-origin',
+    };
+    const visibilityCheckboxes = [...doc.querySelectorAll(SELECTOR_VISIBILITY_CHECKBOXES)];
 
-        doc.querySelector('#location_update_visibility_data_location').value = event.target.value;
-        doc.querySelector('#location_update_visibility_data_hidden').checked = !event.target.checked;
+    const onVisibilityUpdated = (event) => {
+        const { target } = event;
+        const { checked: isHidden } = target;
 
-        doc.querySelector('form[name="location_update_visibility_data"]').submit();
-    }
+        target.closest('.ez-checkbox-icon').classList.toggle('is-checked', isHidden);
+    };
+    const handleUpdateError = global.eZ.services.notification.showErrorNotification;
+    const handleUpdateResponse = (event) => (response) => {
+        if (response.status === 200) {
+            onVisibilityUpdated(event);
+            return;
+        }
+
+        throw new Error(response.statusText);
+    };
+    const updateVisibility = (event) => {
+        const { value: locationId, checked: isVisible } = event.target;
+        const isHiddenNumberValue = !isVisible ? 1 : 0;
+        const changeVisibilityLink = window.Routing.generate('ezplatform.location.hide', { locationId, hidden: isHiddenNumberValue });
 
-    [...doc.querySelectorAll('.ez-checkbox-icon__checkbox')].forEach(checkbox => checkbox.addEventListener('change', updateVisibility, false));
+        fetch(changeVisibilityLink, fetchInit)
+            .then(handleUpdateResponse(event))
```

Comment:
Why not just `handleUpdateResponse.bind(null, event)`?

Reply:
I think that this: `handleUpdateResponse(event)` is more clear than this: `handleUpdateResponse.bind(null, event)`.
In my opinion `bind` don't express well intention of applying one argument to the function. Higher order functions are better IMO.

Reply:
It's not a common approach in our codebase to use HoF. I'd stick with current approach and use `.bind()` function instead.

Reply:
OK, this is some argument. 
I accept it, consistency is important, but do you have any other arguments against using HoF in such cases? I am just asking from pure curiosity about good practices. :)

Reply:
There is no point to create some workarounds if something is native in JS. 
This is about using `.bind()` vs using imitation of `bind`.